src/main/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergTableSchemaValidator.java

```diff
@@ -15,19 +20,28 @@ public IcebergTableSchemaValidator(SnowflakeConnectionService snowflakeConnectio
   /**
    * Ensure that table exists and record_metadata column is of type OBJECT().
    *
-   * <p>TODO SNOW-1658914 - write a test for table with record_metadata schema altered by the
-   * connector
+   * @param tableName table to be validated
+   * @param role role used for validation
    */
   public void validateTable(String tableName, String role) {
-    // TODO - plug into connector startup
-    if (!snowflakeConnectionService.tableExist(tableName)) {
-      throw SnowflakeErrors.ERROR_0032.getException("table_not_found");
+    List<DescribeTableRow> columns =
+        snowflakeConnectionService
+            .describeTable(tableName)
+            .orElseThrow(() -> SnowflakeErrors.ERROR_0032.getException("table_not_found"));
+
+    DescribeTableRow metadata =
+        columns.stream()
+            .filter(c -> Objects.equals(c.getColumn(), TABLE_COLUMN_METADATA))
+            .findFirst()
+            .orElseThrow(
+                () -> SnowflakeErrors.ERROR_0032.getException("record_metadata_not_found"));
+
+    if (!metadata.getType().startsWith("OBJECT")) {
```

Comment:
Why did you use `OBJECT` here? Could you please extract it to a separate variable with a name explaining the meaning of its value?

Reply:
this is the type that;s gonna be required for record_metadata column for iceberg.

But that's a good idea to have a class with iceberg types but not necessary in this PR, we can refactor it later

Reply:
`OBJECT` is a Snowflake type (like `OBJECT()` or `OBJECT(id VARCHAR, age NUMBER)`. I will extract it to `SF_STRUCTURED_OBJECT` to be more precise on that.